app/src/main/java/com/lightricks/feedexercise/FeedApplication.kt

```diff
@@ -0,0 +1,23 @@
+package com.lightricks.feedexercise
+
+import android.app.Application
+import com.lightricks.feedexercise.data.FeedRepository
+import com.lightricks.feedexercise.data.FeedRepositoryImpl
+import com.lightricks.feedexercise.database.AppDatabase
+import com.lightricks.feedexercise.network.FeedApiService
+
+class FeedApplication : Application() {
+
+    private lateinit var appDatabase : AppDatabase
+    private lateinit var feedApiService: FeedApiService
+    lateinit var feedRepository: FeedRepository
+
+    override fun onCreate() {
+        super.onCreate()
+        appDatabase = AppDatabase.getInstance(this)
+        feedApiService = FeedApiService.feedApiService
+        feedRepository = FeedRepositoryImpl(feedApiService, appDatabase.feedDao())
```

Comment:
This is legitimate. But in general, initialising all services in `Application` causes the app to launch more slowly. 
The preferred way is to use Kotlin `objects` which are singletons, because they are created only when you use them for the first time.

Reply:
Well, first, all the singleton now use the technique you mentioned.
Second, personally I believe that in small project that have no DI implemented, the best thing to do is to use `Application` to simulate DI. With this approach in mind, when the project evolve to the point where you want to start using DI, all you need to do is move the injection from `Application` to the appropriate place.